internal/clients/oidc.go

```diff
@@ -44,11 +45,15 @@ func (oc *OAuth2OIDCClient) StartSetup() {
 	var err error
 	ctx := context.Background()
 
-	oc.provider, err = oidc.NewProvider(ctx, oc.configuration.OIDProviderURL)
-	if err != nil {
-		oc.logger.Fatalf("✖️ OIDC Provider setup failed. Error: %v", err)
+	for {
+		oc.provider, err = oidc.NewProvider(ctx, oc.configuration.OIDProviderURL)
+		if err == nil {
+			oc.logger.Info("✔️ Connected to OIDC Provider")
+			break
+		}
+		oc.logger.Errorf("✖️ OIDC Provider setup failed. Error: %v", err)
+		time.Sleep(time.Second * 5)
```

Comment:
do we really need this?

Reply:
maybe we were able to optimize this 🤔 
because it only connects on the second try...

Reply:
we solved it in PR: https://github.com/ydataai/go-core/pull/50